src/components/Nav/index.tsx

```diff
@@ -62,52 +63,48 @@ const Nav: FC<IProps> = ({ path }) => {
           </Link>
           {/* Desktop */}
           <Flex
-            justifyContent="space-between"
-            w="100%"
-            display={{ base: "none", lg: "flex" }}
+            w="full"
+            justifyContent={{ base: "flex-end", lg: "space-between" }}
             ml={{ base: 3, xl: 8 }}
           >
-            <Menu path={path} sections={linkSections} />
+            <Menu hideBelow="lg" path={path} sections={linkSections} />
             <Flex
               alignItems="center"
               justifyContent="space-between"
-              gap={{ base: 1, xl: 0 }}
+              gap={{ base: 2, xl: 4 }}
             >
-              <Search ref={searchRef} />
-              <IconButton
-                aria-label={
-                  isDarkTheme ? "Switch to Light Theme" : "Switch to Dark Theme"
-                }
-                icon={<Icon as={isDarkTheme ? MdWbSunny : MdBrightness2} />}
-                variant="icon"
-                size="sm"
-                fontSize="2xl"
-                ms={{ xl: 2 }}
-                _hover={{ color: "primary.base" }}
-                onClick={toggleColorMode}
+              <Search {...searchModalDisclosure} />
+              {/* Mobile */}
+              <MobileNavMenu
+                {...mobileNavProps}
+                hideFrom="lg"
+                toggleSearch={searchModalDisclosure.onOpen}
+                drawerContainerRef={navWrapperRef}
               />
-              <ButtonLink
-                to={`/languages/${fromPageParameter}`}
-                variant="icon"
-                px={{ base: 1, xl: 1.5 }}
-                size="sm"
-                fontSize="md"
-              >
-                <Icon as={MdLanguage} fontSize="2xl" />
-                <Text as="span" pl={2}>
-                  <Box as="span" hideBelow="lg">
-                    {t("languages")}
-                  </Box>{" "}
-                  {i18n.language.toUpperCase()}
-                </Text>
-              </ButtonLink>
+              <HStack spacing={2} hideBelow="lg">
+                <Button
+                  aria-label={
+                    isDarkTheme
+                      ? "Switch to Light Theme"
+                      : "Switch to Dark Theme"
+                  }
+                  variant="secondaryGhost"
+                  px={1.5}
```

Comment:
I'd like to have your 👀 on this PR @TylerAPfledderer, especially with the following changes:

1st change is in the `Button` component, where I had to remove the `sx` inner logic because otherwise, I wasn't able to override the `px` prop. Not sure exactly why (I think it is because the `sx` prop has more priority than the style props).

2nd change is a consequence of the 1st one. Since I removed the usage of `useStyleConfig` and `sx` from the Button component (and ButtonLink), I started to see the warning about the `isSecondary` flag. I ended up removing it and creating 2 new variants to cover the secondary cases. Not ideal I think, but it worked and is simple to understand.

Let me know what do you think about these changes or if you have any concerns. Also, feel free to test the rest of the changes as well and leave any comments :)

Reply:
To the first, I can see where the issue would happen, so this is definitely cleaner! 😄 

As to the second, it is certainly a matter of preference. I guess here it is readability versus overkill? I'm good with this though! Indeed still easy to roll with.

Reply:
Thanks @TylerAPfledderer 

This PR is a draft because I wanted to first check with you about it and see if you had a workaround to keep the same convention you built (with simpler variants + the `isSecondary` flag). Tbh I don't like having variants called `"secondaryOutline"` hahaha.

So, let me know if you prefer to hold on a bit here and see if you can add the necessary changes to make this work with the original convention you did.

Reply:
Awesome thanks!

LMK if you can add commits to this PR (I doubt it) or create a new PR off this branch and we can merge it back.